redash/query_runner/snowflake.py

```diff
@@ -96,14 +104,26 @@ def _get_connection(self):
             else:
                 host = "{}.snowflakecomputing.com".format(account)
 
-        connection = snowflake.connector.connect(
-            user=self.configuration["user"],
-            password=self.configuration["password"],
-            account=account,
-            region=region,
-            host=host,
-            application="Redash/{} (Snowflake)".format(__version__.split("-")[0]),
-        )
+        params = {
+            "user": self.configuration["user"],
+            "account": account,
+            "region": region,
+            "host": host,
+            "application": "Redash/{} (Snowflake)".format(__version__.split("-")[0])
+        }
+
+        if self.configuration.__contains__("password"):
```

Comment:
It looks this is always true even if password is empty string.

Reply:
@adrianoesch 

self.configuration["password"] can be empty string if you set and save the password and clear the password.
Similar issue was reported as https://github.com/getredash/redash/issues/7486 .

Also, I think "__contains__" special method is not indented to be directly called.

How about changing to like below ?
```
if self.configuration.get("password"):
```